addons/website/models/website.py

```diff
@@ -384,11 +385,45 @@ def set_images(images):
         requested_pages = list(pages_views.keys())
         requested_pages.append('homepage')
         params = {
-            'theme': kwargs.get('theme_name'),
+            'theme': theme_name,
             'pages': requested_pages,
         }
         custom_resources = self._website_api_rpc('/api/website/1/configurator/custom_resources/%s' % kwargs.get('industry_id'), params)
 
+        # Update CTA
+        cta_data = False
+        if kwargs.get('website_purpose') == 'schedule_appointments':
+            domain = [('name', '=', 'website_calendar'), ('state', '=', 'installed')]
+            is_website_calendar_installed = bool(self.env['ir.module.module'].search(domain))
+            cta_data = is_website_calendar_installed and {'text': _('Schedule an appointment'), 'href': '/calendar'}
```

Comment:
Some themes put button text in uppercase. We don't take this into account here. It is particularly a problem for paptic theme were two buttons are side by side in the cover snippet. Having one button with uppercase text and one without is not beautiful. I don't see better solution than also hard-coding which theme has/hasn't uppercase text. Any idea?

Reply:
I think that, when in need to uppercase a button, the 90% of the users will just **type** uppercase...

Pro-users may appreciate a general option that force the behaviour, but such option would likely block from ad-hoc customisations when a lowercase one is needed (except if we add the option per-button too, increasing complexity even more...).

Uppercased buttons for snippets should be a "design suggestion" IMO, not blocking the user from lowercase if he wants to. It's the approach I used for Paptic.
When Paptic is "default", indeed, the combination of text size + font weight + font "style" for buttons leads to uppercase to work better...

The problem comes with the Apps' button as @qsm-odoo said, but are these buttons supposed to be editable?
The "ADD TO CART" button is uppercase by default and I can't lowercase because of the `text-uppercase` class  🤔

Definitively something to check @sbuhl 
![image](https://user-images.githubusercontent.com/8558351/120449888-234eb600-c390-11eb-9bbf-a7c8d5d74d99.png)

Reply:
Thanks for the feedback

> but such option would likely block from ad-hoc customisations when a lowercase one is needed 

That, is a good argument against the option indeed

> The "ADD TO CART" button is uppercase by default and I can't lowercase because of the text-uppercase class thinking

I would fix that indeed by removing the class and typing the text in uppercase letters (as the text-uppercase class option would be too "complex")

---

So @smottet, to investigate around this idea:

> Anyways, that would be for a master task anyway. For 14.3, can't you just check if the original text in the current snippet is uppercased and uppercase yours if that is the case (damn that's a lot of "case" in one sentence grimacing)? Not really pretty but shrug

which may not be really compatible with our "define a variable in a t-set outside of the snippet" solution. Other potential solution: creating a class which indicates to the editor that the text the element contains should be manually uppercased on drop. That way paptic could add that class on buttons it wants, the configurator can override the text with lowercased words and they would become uppercased on drop in the page.